scaaml/capture/input_generators/attack_point_iterator.py

```diff
@@ -71,3 +81,62 @@ def __len__(self) -> int:
 
     def __iter__(self):
         return iter({self._name: value} for value in self._values)
+
+
+class AttackPointIteratorInternalBalancedGenerator(
+        AttackPointIteratorInternalBase):
+    """
+    Attack point iterator class that iterates over the balanced generator.
+    """
+
+    def __init__(self,
+                 operation: str,
+                 name: str,
+                 length: int,
+                 bunches: int = 1,
+                 elements: int = 256) -> None:
+        """Initialize the balanced kwargs to iterate."""
+        self._operation = operation
```

Comment:
Why are we saving the operation name? We know what it should be. So either make sure it has the expected value:
`assert operation == "balanced_generator"` or ignore it `del operation`.